tests/runtime/test_bash.py

```diff
@@ -0,0 +1,744 @@
+
+import asyncio
+import os
+import tempfile
+import time
+
+import pytest
+from pytest import TempPathFactory
+
+from openhands.core.config import AppConfig, SandboxConfig, load_from_env
+from openhands.core.logger import openhands_logger as logger
+from openhands.events import EventStream
+from openhands.events.action import CmdRunAction
+from openhands.events.observation import CmdOutputObservation
+from openhands.runtime.client.runtime import EventStreamRuntime
+from openhands.runtime.plugins import AgentSkillsRequirement, JupyterRequirement
+from openhands.runtime.runtime import Runtime
+from openhands.storage import get_file_store
+
+
+@pytest.fixture(autouse=True)
+def print_method_name(request):
+    print('\n########################################################################')
+    print(f'Running test: {request.node.name}')
+    print('########################################################################')
+    yield
+
+
+@pytest.fixture
+def temp_dir(tmp_path_factory: TempPathFactory) -> str:
+    return str(tmp_path_factory.mktemp('test_runtime'))
+
+
+TEST_RUNTIME = os.getenv('TEST_RUNTIME', 'eventstream')
+PY3_FOR_TESTING = '/openhands/miniforge3/bin/mamba run -n base python3'
+
+
+# Depending on TEST_RUNTIME, feed the appropriate box class(es) to the test.
+def get_box_classes():
+    runtime = TEST_RUNTIME
+    if runtime.lower() == 'eventstream':
+        return [EventStreamRuntime]
+    else:
+        raise ValueError(f'Invalid runtime: {runtime}')
+
+
+# This assures that all tests run together per runtime, not alternating between them,
+# which cause errors (especially outside GitHub actions).
+@pytest.fixture(scope='module', params=get_box_classes())
+def box_class(request):
+    time.sleep(2)
+    return request.param
+
+
+# TODO: We will change this to `run_as_user` when `ServerRuntime` is deprecated.
+# since `EventStreamRuntime` supports running as an arbitrary user.
+@pytest.fixture(scope='module', params=[True, False])
+def run_as_openhands(request):
+    time.sleep(1)
+    return request.param
+
+
+@pytest.fixture(scope='module', params=[True, False])
+def enable_auto_lint(request):
+    time.sleep(1)
+    return request.param
+
+
+@pytest.fixture(scope='module', params=None)
+def container_image(request):
+    time.sleep(1)
+    env_image = os.environ.get('SANDBOX_CONTAINER_IMAGE')
+    if env_image:
+        request.param = env_image
+    else:
+        if not hasattr(request, 'param'):  # prevent runtime AttributeError
+            request.param = None
+        if request.param is None:
+            request.param = request.config.getoption('--container-image')
+        if request.param is None:
+            request.param = pytest.param(
+                'nikolaik/python-nodejs:python3.11-nodejs22',
+                'python:3.11-bookworm',
+                'node:22-bookworm',
+                'golang:1.23-bookworm',
+            )
+    print(f'Container image: {request.param}')
+    return request.param
```

Comment:
Should we put these into a separate contest file?https://docs.pytest.org/en/stable/reference/fixtures.html#conftest-py-sharing-fixtures-across-multiple-files

Reply:
Makes total sense, will look into it